vpr/src/place/place_delay_model.h

```diff
@@ -6,6 +6,20 @@
 #include "vpr_types.h"
 #include "router_delay_profiling.h"
 
+#ifndef __has_attribute
+#    define __has_attribute(x) 0 // Compatibility with non-clang compilers.
+#endif
+
+#if defined(COMPILER_GCC) && defined(NDEBUG)
+#    define ALWAYS_INLINE inline __attribute__((__always_inline__))
+#elif defined(COMPILER_MSVC) && defined(NDEBUG)
+#    define ALWAYS_INLINE __forceinline
+#elif __has_attribute(always_inline)
+#    define ALWAYS_INLINE __attribute__((always_inline)) // clang
+#else
+#    define ALWAYS_INLINE inline
+#endif
+
```

Comment:
I wonder if it would be better to use [C++ attributes](https://en.cppreference.com/w/cpp/language/attributes) (available since C++11) as a cleaner way to specify this.

For example:
```cpp
[[gnu::always_inline]] friend bool operator<(const t_override& lhs, const t_override& rhs) {
```

That would avoid having to define the ALWAYS_INLINE macro in a cleaner cross-compiler way.

@mithro Any thoughts on this?